### PR TITLE
chore: update deprecated usage of coder_workspace.owner* fields

### LINKED ISSUE #13382: update example templates to use `data.coder_workspace_owner`
After https://github.com/coder/terraform-provider-coder/releases/tag/v0.23.0, some owner-related fields of `data.coder_workspace` are deprecated and it is recommended to use the newer [`coder_workspace_owner`](https://registry.terraform.io/providers/coder/coder/latest/docs/data-sources/workspace_owner) data source.

An example usage,
https://github.com/coder/coder/blob/a1d3b82dd128838dd9a1edc3d8bca247e3618a52/examples/templates/docker/main.tf#L46-L51

### PR DESCRIPTION
**Pull request by johnstcn**
Per https://github.com/coder/terraform-provider-coder/releases/tag/v0.23.0

Performs a mechanical rename of existing usage deprecated fields in the latest version of the `coder/coder` provider.

Closes #13382

### CODE REVIEW

File: examples/templates/azure-windows/main.tf [hunk 1]
```diff
@@ -17,6 +17,7 @@ provider "coder" {
 }
 
 data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
```
Comment (spikecurtis) [inline examples/templates/azure-windows/main.tf:20]: unused

File: examples/templates/scratch/main.tf [hunk 2]
```diff
@@ -9,6 +9,7 @@ terraform {
 data "coder_provisioner" "me" {}
 
 data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
```
Comment (spikecurtis) [inline examples/templates/scratch/main.tf:12]: unused

File: scaletest/terraform/k8s/coder.tf [hunk 3]
```diff
@@ -300,6 +300,7 @@ resource "local_file" "kubernetes_template" {
     }
 
     data "coder_workspace" "me" {}
+data "coder_workspace_owner" "me" {}
```
Comment (spikecurtis) [inline scaletest/terraform/k8s/coder.tf:303]: incorrect indentation

File: site/src/modules/templates/TemplateFiles/TemplateFiles.stories.tsx [hunk 4]
```diff
@@ -8,7 +8,7 @@ const exampleFiles = {
...
...main.name\n  # Uses lower() to avoid Docker restriction on container names.\n  name = "coder-${data.coder_workspace_owner.me.name}-${lower(data.coder_workspace.me.name)}"\n  # Hostname makes the shell more user friendly: coder@my-workspace:~$\n  hostname = data.coder_workspace.me.name\n  # Use the docker gateway if the access URL is 127.0.0.1\n  entrypoint = ["sh", "-c", replace(coder_agent.main.init_script, "/localhost|127\\\\.0\\\\.0\\\\.1/", "host.docker.internal")]\n  env        = ["CODER_AGENT_TOKEN=${coder_agent.main.token}"]\n  host {\n    host = "host.docker.internal"\n    ip   = "host-gateway"\n  }\n  volumes {\n    container_path = "/home/${local.username}"\n    volume_name    = docker_volume.home_volume.name\n    read_only      = false\n  }\n  # Add labels in Docker to keep track of orphan resources.\n  labels {\n    label = "coder.owner"\n    value = data.coder_workspace_owner.me.name\n  }\n  labels {\n    label = "coder.owner_id"\n    value = data.coder_workspace_owner.me.id\n  }\n  labels {\n    label = "coder.workspace_id"\n    value = data.coder_workspace.me.id\n  }\n  labels {\n    label = "coder.workspace_name"\n    value = data.coder_workspace.me.name\n  }\n}\n',
```
Comment (spikecurtis) [inline site/src/modules/templates/TemplateFiles/TemplateFiles.stories.tsx:11]: missing `data "coder_workspace_owner" "me"`

Comment (johnstcn) [reply site/src/modules/templates/TemplateFiles/TemplateFiles.stories.tsx]: Well spotted!

**Review by spikecurtis — APPROVED** [review]

**Pull request merged**